pkgs/development/compilers/halide/default.nix

```diff
@@ -2,55 +2,38 @@
 , lib
 , fetchFromGitHub
 , cmake
-, libpng
-, libjpeg
 , mesa
-, eigen
-, openblas
-, blas
-, lapack
 }:
 
-assert blas.implementation == "openblas" && lapack.implementation == "openblas";
-
 llvmPackages.stdenv.mkDerivation rec {
   pname = "halide";
-  version = "10.0.0";
+  version = "14.0.0";
 
   src = fetchFromGitHub {
     owner = "halide";
     repo = "Halide";
     rev = "v${version}";
-    sha256 = "0il71rppjp76m7zd420siidvhs76sqiq26h60ywk812sj9mmgxj6";
+    sha256 = "sha256-/7U2TBcpSAKeEyWncAbtW6Vk/cP+rp1CXtbIjvQMmZA=";
   };
 
-  # clang fails to compile intermediate code because
-  # of unused "--gcc-toolchain" option
-  postPatch = ''
-    sed -i "s/-Werror//" src/CMakeLists.txt
-  '';
-
-  cmakeFlags = [ "-DWARNINGS_AS_ERRORS=OFF" "-DWITH_PYTHON_BINDINGS=OFF" ];
+  cmakeFlags = [
+    "-DWARNINGS_AS_ERRORS=OFF"
+    "-DWITH_PYTHON_BINDINGS=OFF"
+    "-DTARGET_WEBASSEMBLY=OFF"
+  ];
 
   # To handle the lack of 'local' RPATH; required, as they call one of
   # their built binaries requiring their libs, in the build process.
   preBuild = ''
     export LD_LIBRARY_PATH="$(pwd)/src''${LD_LIBRARY_PATH:+:}$LD_LIBRARY_PATH"
   '';
 
-  # Note: only openblas and not atlas part of this Nix expression
-  # see pkgs/development/libraries/science/math/liblapack/3.5.0.nix
-  # to get a hint howto setup atlas instead of openblas
   buildInputs = [
     llvmPackages.llvm
     llvmPackages.lld
     llvmPackages.openmp
     llvmPackages.libclang
-    libpng
-    libjpeg
     mesa
```

Comment:
You are sure mesa does not go into nativeBuildInputs?

Reply:
I'm not sure why mesa is here at all? It's not a requirement to build Halide. It's probably a requirement to build apps using Halide's OpenGL Compute backend, though.

Reply:
It would be interesting to have a wrapper for halide dependent projects, but we have to think about how to best design this wrapper to fit the needs of a specific app first.